xlive/H2ConsoleCommands.cpp

```diff
@@ -230,7 +230,9 @@ void ConsoleCommands::handle_command(std::string command) {
 			BYTE maxPlayersNumber = *(BYTE*)(lobby_globals + 0x4C80);
 
 			if (isNum(cstr)) {
-				int maxPlayersSet = atoi(cstr);
+				delete[] cstr;
```

Comment:
you should use a try finally here, in the finally block you can do the delete in there and outside the try do char *cstr = NULL;
inside the try, do cstr = new char[firstArg.length() + 1];

then in finally, do if (cstr != null) delete[] cstr;